### PR TITLE
Fixes for sparse projection

### PR DESCRIPTION
**Pull request by fgrosa**
to be squashed and merged

### CODE REVIEW

File: utils/TaskFileLoader.py [hunk 1]
```diff
@@ -16,15 +17,23 @@ def LoadSparseFromTask(infilename, inputCfg):
...
     - list of sparses with generated quantities for prompt and FD D mesons (only if MC)
     '''
     print('Loading THnSparses from file', infilename)
     infileData = TFile(infilename)
     indirData = infileData.Get(inputCfg['dirname'])
     inlistData = indirData.Get(inputCfg['listname'])
+
+    if inputCfg['sparsenameAll'] == inputCfg['sparsenamePrompt'] or \
+        inputCfg['sparsenameAll'] == inputCfg['sparsenameFD']:
+        print('ERROR: do not use the same object for different spareses, this gives an error when merged! Exit')
```
Comment (fcatalan92) [inline utils/TaskFileLoader.py:31]: Typo `spareses` -> `sparses`

**Review by fcatalan92 — CHANGES_REQUESTED** [review]

File: utils/TaskFileLoader.py [hunk 2]
```diff
@@ -16,15 +17,23 @@ def LoadSparseFromTask(infilename, inputCfg):
...
+        sys.exit()
+
     sparses, sparsesGen = {}, {}
-    sparses['RecoAll'] = inlistData.FindObject(inputCfg['sparsenameAll'])
+    if inputCfg['sparsenameAll'] and inlistData.FindObject(inputCfg['sparsenameAll']):
```
Comment (fcatalan92) [inline utils/TaskFileLoader.py:35]: This check for the `FindObject` is not done for the other histograms. We can either have a check for all or none

Comment (fgrosa) [reply utils/TaskFileLoader.py]: Done for all in b36924e

File: utils/TaskFileLoader.py (hunk 1 above)
Comment (fgrosa) [reply utils/TaskFileLoader.py]: Done

File: utils/TaskFileLoader.py [hunk 3]
```diff
@@ -228,6 +299,12 @@ def LoadListFromTaskV2(infilename, dirname, listname):
     print('Loading TList from file', infilename)
     infileData = TFile(infilename)
     indirData = infileData.Get(dirname)
+    if not indirData:
+        print(f'Directory {dirname} not found!')
+        return None, None
```
Comment (fcatalan92) [inline utils/TaskFileLoader.py:304]: Too many return values

**Review by fcatalan92 — CHANGES_REQUESTED** [review]
Please read again the code, not sure to have catched them all!

**Review by fcatalan92 — APPROVED** [review]

**Comment by fgrosa** [comment]
Hi @fcatalan92 thanks a lot for checking. I did it in a hurry but now should be ok

**Pull request merged**